From 901e394913c875ec1884af62a59b9323aae2c425 Mon Sep 17 00:00:00 2001
From: Alexander Schulze <alexschulze@chromium.org>
Date: Thu, 1 Jun 2023 13:01:10 +0200
Subject: [PATCH] Update V8 DEPS (manual, android build bug)

This also adds the cpu_features gn file from
https://crrev.com/c/4545271.

DEPS are updated to the ones determined by the auto-roller in
https://crrev.com/c/4576153.

Bug: chromium:1450518
Change-Id: I8e41e43c64c2afd31ab22ba0007f5a5078499135
No-Tree-Checks: true
Reviewed-on: https://chromium-review.googlesource.com/c/v8/v8/+/4577282
Reviewed-by: Michael Achenbach <machenbach@chromium.org>
Commit-Queue: Alexander Schulze <alexschulze@chromium.org>
Reviewed-by: Michael Lippautz <mlippautz@chromium.org>
Cr-Commit-Position: refs/heads/main@{#87991}
---
 .gitignore                         |  2 +
 DEPS                               |  6 +--
 third_party/cpu_features/BUILD.gn  | 61 ++++++++++++++++++++++++++++++
 third_party/cpu_features/OWNERS    |  1 +
 third_party/cpu_features/README.v8 | 14 +++++++
 5 files changed, 81 insertions(+), 3 deletions(-)
 create mode 100644 third_party/cpu_features/BUILD.gn
 create mode 100644 third_party/cpu_features/OWNERS
 create mode 100644 third_party/cpu_features/README.v8

diff --git a/.gitignore b/.gitignore
index 1687038dc7..65fe25d1b8 100644
--- a/.gitignore
+++ b/.gitignore
@@ -64,6 +64,8 @@
 /test/wasm-spec-tests/tests.tar.gz
 /third_party/*
 !/third_party/antlr4
+!/third_party/cpu_features
+/third_party/cpu_features/src
 !/third_party/inspector_protocol
 !/third_party/jsoncpp
 /third_party/jsoncpp/source
diff --git a/DEPS b/DEPS
index 0e9d1648a1..a30aaf8af7 100644
--- a/DEPS
+++ b/DEPS
@@ -166,11 +166,11 @@ deps = {
   'test/test262/data':
     Var('chromium_url') + '/external/github.com/tc39/test262.git' + '@' + 'beb4f26eb4e6f6f2bf71c8441521aaa950e62052',
   'third_party/android_ndk': {
-    'url': Var('chromium_url') + '/android_ndk.git' + '@' + '8388a2be5421311dc75c5f937aae13d821a27f3d',
+    'url': Var('chromium_url') + '/android_ndk.git' + '@' + '310956bd122ec2b96049f8d7398de6b717f3452e',
     'condition': 'checkout_android',
   },
   'third_party/android_platform': {
-    'url': Var('chromium_url') + '/chromium/src/third_party/android_platform.git' + '@' + 'f312145c4191affc66e7a1d46194f0d6c9dec438',
+    'url': Var('chromium_url') + '/chromium/src/third_party/android_platform.git' + '@' + '9f29aca7dd5ad56865863e5810ffb96756c241c3',
     'condition': 'checkout_android',
   },
   'third_party/android_sdk/public': {
@@ -277,7 +277,7 @@ deps = {
       'condition': 'checkout_android',
   },
   'third_party/zlib':
-    Var('chromium_url') + '/chromium/src/third_party/zlib.git'+ '@' + '337322d10f407b25671768f092e55f306c1a1101',
+    Var('chromium_url') + '/chromium/src/third_party/zlib.git'+ '@' + '982b036a257e63240142358a0fce0dbfdd40d9cb',
   'tools/clang':
     Var('chromium_url') + '/chromium/src/tools/clang.git' + '@' + '3ad8228c907ab5af58f49f4d093d8aef7768c198',
   'tools/luci-go': {
diff --git a/third_party/cpu_features/BUILD.gn b/third_party/cpu_features/BUILD.gn
new file mode 100644
index 0000000000..527cf2c9aa
--- /dev/null
+++ b/third_party/cpu_features/BUILD.gn
@@ -0,0 +1,61 @@
+# Copyright 2023 The Chromium Authors. All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+config("cpu_features_config") {
+  cflags = [ "-Wno-unused-function" ]
+  defines = [
+    "STACK_LINE_READER_BUFFER_SIZE=1024",
+    "HAVE_STRONG_GETAUXVAL",
+  ]
+  include_dirs = [ "src/include" ]
+}
+
+config("ndk_compat_headers") {
+  include_dirs = [ "src/ndk_compat" ]
+}
+
+source_set("cpuinfo") {
+  sources = [
+    "src/src/copy.inl",
+    "src/src/define_introspection.inl",
+    "src/src/define_introspection_and_hwcaps.inl",
+    "src/src/equals.inl",
+    "src/src/filesystem.c",
+    "src/src/hwcaps.c",
+    "src/src/stack_line_reader.c",
+    "src/src/string_view.c",
+  ]
+  if (current_cpu == "x86" || current_cpu == "x64") {
+    sources += [
+      "src/src/impl_x86__base_implementation.inl",
+      "src/src/impl_x86_freebsd.c",
+      "src/src/impl_x86_linux_or_android.c",
+      "src/src/impl_x86_macos.c",
+      "src/src/impl_x86_windows.c",
+    ]
+  } else if (current_cpu == "arm") {
+    sources += [ "src/src/impl_arm_linux_or_android.c" ]
+  } else if (current_cpu == "arm64") {
+    sources += [ "src/src/impl_aarch64_linux_or_android.c" ]
+  } else if (current_cpu == "mips") {
+    sources += [ "src/src/impl_mips_linux_or_android.c" ]
+  } else if (current_cpu == "ppc") {
+    sources += [ "src/src/impl_ppc_linux.c" ]
+  } else if (current_cpu == "riscv64") {
+    sources += [ "src/src/impl_riscv_linux.c" ]
+  } else {
+    error("Missing definition for architecture: $current_cpu")
+  }
+  configs += [ ":cpu_features_config" ]
+}
+
+source_set("ndk_compat") {
+  sources = [
+    "src/ndk_compat/cpu-features.c",
+    "src/ndk_compat/cpu-features.h",
+  ]
+  configs += [ ":cpu_features_config" ]
+  public_configs = [ ":ndk_compat_headers" ]
+  deps = [ ":cpuinfo" ]
+}
diff --git a/third_party/cpu_features/OWNERS b/third_party/cpu_features/OWNERS
new file mode 100644
index 0000000000..09e0096a2e
--- /dev/null
+++ b/third_party/cpu_features/OWNERS
@@ -0,0 +1 @@
+file:../../INFRA_OWNERS
diff --git a/third_party/cpu_features/README.v8 b/third_party/cpu_features/README.v8
new file mode 100644
index 0000000000..b1a8ff7390
--- /dev/null
+++ b/third_party/cpu_features/README.v8
@@ -0,0 +1,14 @@
+Name: cpu_features
+Short Name: cpu_features
+URL: https://github.com/google/cpu_features
+Version: v0.8.0
+Date: 2023/05/17
+License: Apache 2.0
+License File: src/LICENSE
+Security Critical: Yes
+
+Description:
+cpu_features is a library to retrieve CPU features at runtime. It is used to
+make decisions about performance optimization and features a drop-in replacement
+for Android's cpu-features.h. It is Android's recommended replacement for libraries
+that utilize cpu-features.h.
-- 
2.35.1

